### PR TITLE
Fix comments inside OR pattern on case list

### LINKED ISSUE #559: poor formatting of comments between switch cases
Consider:

```
let _ =
  switch x {
  | None
  | Some _ => ()
  };
```

If a comment is added between the cases:

```
let _ =
  switch x {
  | None
  /* between */
  | Some _ => ()
  };
```

then refmt is very heavy-handed:

```
let _ =
  switch x {
  | None
  | /* between */
    Some _ =>
    ()
  };
```

It would be good if the `|` and the pattern `Some _` weren't split up, and if the line breaking didn't change.

### PR DESCRIPTION
**Pull request by tadeuzagallo**
Fixes #559 

I also had to update a couple other tests, where comments on or patterns would be pushed to the next line, but I think it has the correct behaviour now.

### CODE REVIEW

**Comment by SanderSpies** [comment]
You are on a roll! Nice.

File: formatTest/unit_tests/expected_output/wrappingTest.re [hunk 1]
```diff
@@ -2216,7 +2216,8 @@ let blah =
 let blahCurriedX x =>
   fun
   /* Comment before first bar */
-  | /* Comment between first bar and OR pattern */ Red x
+  /* Comment between first bar and OR pattern */
+  | Red x
```
Comment (tadeuzagallo) [inline formatTest/unit_tests/expected_output/wrappingTest.re]: This is probably the most questionable change, but should be easy to fix if the original is preferred.

Comment (SanderSpies) [inline formatTest/unit_tests/expected_output/wrappingTest.re]: I would prefer this to be fixed, as the input is:
```
  /* Comment before first bar */
  | /* Comment between first bar and OR pattern */
    (Red x | Black x | Green x) => 1
```

Comment (tadeuzagallo) [inline formatTest/unit_tests/expected_output/wrappingTest.re]: Now it should look like

```swift
/* Comment before first bar */
| /* Comment between first bar and OR pattern */
  Red x
| Black x
| Green x => 1
```

Is it ok?

**Comment by yunxing** [comment]
Should be fixed along with the new comments interleaving code.

**Pull request closed without merging**